File: packages/java/src/main/java/com/tesote/sdk/models/BatchSummary.java
package com.tesote.sdk.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public record BatchSummary(
        @JsonProperty("batch_id") String batchId,
        @JsonProperty("total_orders") Integer totalOrders,
        @JsonProperty("total_amount_cents") Long totalAmountCents,
        @JsonProperty("amount_currency") String amountCurrency,
        @JsonProperty("statuses") Map<String, Integer> statuses,
        @JsonProperty("batch_status") String batchStatus,
        @JsonProperty("created_at") String createdAt,
        @JsonProperty("orders") List<TransactionOrder> orders
) {}

File: packages/php/src/Models/BatchSummary.php
<?php

declare(strict_types=1);

namespace Tesote\Sdk\Models;

/** Response from GET /v2/accounts/{id}/batches/{batch_id}. */
final class BatchSummary
{
    /**
     * @param array<string, int>     $statuses
     * @param list<TransactionOrder> $orders
     */
    public function __construct(
        public readonly string $batchId,
        public readonly int $totalOrders,
        public readonly int $totalAmountCents,
        public readonly string $amountCurrency,
        public readonly array $statuses,
        public readonly string $batchStatus,
        public readonly string $createdAt,
        public readonly array $orders,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        $statuses = [];
        foreach ((is_array($data['statuses'] ?? null) ? $data['statuses'] : []) as $status => $count) {
            $statuses[(string) $status] = (int) $count;
        }
        $orders = [];
        foreach ((is_array($data['orders'] ?? null) ? $data['orders'] : []) as $entry) {
            if (is_array($entry)) {
                $orders[] = TransactionOrder::fromArray($entry);
            }
        }

        return new self(
            batchId: (string) ($data['batch_id'] ?? ''),
            totalOrders: (int) ($data['total_orders'] ?? 0),
            totalAmountCents: (int) ($data['total_amount_cents'] ?? 0),
            amountCurrency: (string) ($data['amount_currency'] ?? ''),
            statuses: $statuses,
            batchStatus: (string) ($data['batch_status'] ?? ''),
            createdAt: (string) ($data['created_at'] ?? ''),
            orders: $orders,
        );
    }
}
